### PR TITLE
[ASR][Breaking Change] Update signature of Hypothesis alignments

### PR DESCRIPTION
**Pull request by titu1994**
# What does this PR do ?

Updates signature of Hypothis.alignments such that it now contains a tuple (logprob, label), such that CTC, RNNT greedy and RNNT beam search all yield the logprob tensor of shape (V+1,) where V is the vocab size (and last index is CTC/RNNT Blank). This can be used to compute beam search results.

For CTC Models, hypothesis.alignments now returns a tuple of tensors (logp, labels) rather than just logp to match the RNNT signature.

**Collection**: [ASR]

# Changelog 
- [BREAKING CHANGE] Force `preserve_alignments` flag to preserve a tuple of values (logp, label) for both CTC and RNNT.
- This enables external beam search rescoring to work with CTC and RNNT models.
- Also enables the possibility of RNNT based char/word time stamp calculation
- In totality, enables CTC/RNNT models to compute word confidence.
- Add respective tests

# Usage

```python

model = EncDecRNNTModelBPE.from_pretrained(...)
model.eval()

decoding_cfg. model.cfg.decoding
decoding_cfg.preserve_alignments = True
model.change_decoding_strategy(decoding_cfg)

# For RNNT
hyps, _ = model.transcribe(..., return_hypothesis=True)
# For CTC
hyps = model.transcribe(..., return_hypothesis=True)

# select hypothesis of 1st sample
hyp = hyps[0]

# extract alignments
for t in range(len(hyp.alignments)):
  for u in range(len(hyp.alignments[t])):
    logprob, label = hyp[t][u]
    ....

```

# Before your PR is "Ready for review"
**Pre checks**:
- [x] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [x] Did you write any new necessary tests?
- [x] Did you add or update any necessary documentation?
- [x] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [x] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
- [x] New Feature
- [ ] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

### CODE REVIEW

**Review by krishnacpuvvada — APPROVED** [review]
LGTM
